Lighting/Src/lighting_controller.cpp

```diff
@@ -212,40 +159,289 @@ void LightingController::remove_led_from_cd(uint8_t index,
 	this->domain_leds[domain] &= ~(1 << index);
 }
 
-void LightingController::set_domain_colour(ControlDomain domain,
+void LightingController::set_domain_colour_and_brightness(ControlDomain domain,
 		RGB_colour_t colour, uint8_t brightness) {
 	this->domain_colours[domain] = colour;
 	this->domain_brightness[domain] = brightness;
 }
 
-void LightingController::enable_control_domain(ControlDomain domain) {
-	this->domain_state[domain] = true;	// TODO: have active/disabled booleans
-	for (int i = 0; i < CD_LENGTH; ++i) {
-		if (this->domain_state[i]) { // IF THIS DOMAIN IS ACTIVE
-			for (int j = 0; j < NUM_LEDS; ++j) {
-				if (this->domain_leds[i] & (1 << j)) {
-					this->leds[j].set_led_colour(domain_colours[i],
-							domain_brightness[i]);
+void LightingController::set_domain_brightness(ControlDomain domain, uint8_t brightness) {
+	this->domain_brightness[domain] = brightness;
+}
+
+void LightingController::set_domain_colour(ControlDomain domain, RGB_colour_t colour) {
+	this->domain_colours[domain] = colour;
+}
+
+void LightingController::activate_domain(ControlDomain domain) {
+	if (this->domain_allowed & (1 << domain)) {
+		this->domain_active |= 1 << domain;	// TODO: have active/disabled booleans
+		for (int i = 0; i < CD_LENGTH; ++i) {
```

Comment:
I suppose `i` as a variable iterator is okay here, but it might be fun to use `cd_idx` or something more descriptive?